spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/core/SubscribingOperation.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.integration.core;
+
+import java.util.function.Consumer;
+
+/**
+ * Operations for subscribing to a destination.
+ *
+ * @author Warren Zhu
+ */
+public interface SubscribingOperation<D> {
+
+    /**
+     * Register a message consumer to a given destination.
+     *
+     * @return {@code true} if the consumer was subscribed or {@code false} if it
+     * was already subscribed.
+     */
+    boolean subscribe(String destination, Consumer<Iterable<D>> consumer);
+
+    /**
+     * Un-register a message consumer.
+     *
+     * @return {@code true} if the consumer was un-registered, or {@code false}
```

Comment:
Similar with above, though return value is `void`, but the Javadoc writes the return value is boolean.

Reply:
Typo here, should be boolean